src/test/run-pass/union/union-generic.rs

```diff
@@ -1,30 +1,24 @@
 // run-pass
 #![allow(dead_code)]
-#![allow(unions_with_drop_fields)]
 
 #![feature(untagged_unions)]
 
-union MaybeItem<T: Iterator> {
+union MaybeItem<T: Iterator> where T::Item: Copy {
     elem: T::Item,
```

Comment:
I feel like wrapping this in `ManuallyDrop` more closely preserves the intent of the original test.